opencensus/trace/exporters/file_exporter.py

```diff
@@ -41,12 +41,11 @@ class FileExporter(base.Exporter):
 
     """
 
-    def __init__(self, file_name=DEFAULT_FILENAME,
-                 transport=sync.SyncTransport,
-                 file_mode='w+'):
+    def __init__(self, file_name=DEFAULT_FILENAME, file_mode='w+',
+                 transport=sync.SyncTransport, transport_config=None):
         self.file_name = file_name
-        self.transport = transport(self)
```

Comment:
I wonder why we're initializing the transport here instead of passing the exporter constructor a preconfigured transport. This may be a simpler solution to your problem.

Reply:
@c24t, I suggested this in #249 (pass an object instead), but add a configuration parameter could be interesting for backwards compatibility.

Reply:
I agree with @felippe-mendonca, this is mostly to keep backwards compatibility while giving more options. Maybe we can think of a more global reformat later on.